EIPS/eip-4938.md

```diff
@@ -0,0 +1,57 @@
+---
+eip: 4938
+title: "eth/67: Removal of GetNodeData"
+author: Marius van der Wijden (@MariusVanDerWijden), Felix Lange <fjl@ethereum.org>, Gary Rong <garyrong@ethereum.org>
+discussions-to: https://github.com/ethereum/devp2p/issues/202
+status: Draft
+last-call-deadline: 
+type: Standards Track
+category: Networking
+created: 2022-03-23
+requires: 2464, 2482
+---
+
+## Abstract
+
+The [Ethereum Wire Protocol][eth] defines request and response messages for exchanging data between clients. The `GetNodeData` request retrieves a set of trie nodes or contract code from the state trie by hash. We propose to remove the `GetNodeData` and `NodeData` messages from the wire protocol.
+
+## Motivation
+
+`GetNodeData` and `NodeData` were introduced in protocol version `eth/63` to allow for a sync mode called "fast sync", which downloads the Ethereum state without executing all blocks. The sync algorithm works by requesting all state trie nodes and contract codes by their hash.
+
+Serving `GetNodeData` requests requires clients to store state as a mapping of hashes to trie nodes. Avoiding the need to store such a mapping in the database is the main motivation for removing this request type.
+
+Some client implementations, notably OpenEthereum and Erigon, do not serve `GetNodeData` at this time, because they do not store the state in a compatible way. The Ethereum Wire Protocol should accurately reflect the capabilites of clients, and should not contain messages which are impossible to implement in some clients.
```

Comment:
```suggestion
A client implementation may not serve `GetNodeData` because it does not store the state in a compatible way. The Ethereum Wire Protocol should accurately reflect the capabilities of clients, and should not contain messages which are impossible to implement in some clients.
```
EIPs should be written to be read by someone at an indefinite time in the future, so we shouldn't be discussing current state of current clients but instead abstractly talking about theoretical clients.

Reply:
Alternative suggested text: 

```
At this time, some client implementations cannot serve `GetNodeData` requests because they do not store the state in a compatible way. The Ethereum Wire Protocol...
```